caddyhttp/httpserver/context.go

```diff
@@ -63,6 +63,18 @@ func (c Context) Header(name string) string {
 	return c.Req.Header.Get(name)
 }
 
+// Hostname gets the (remote) hostname of the client making the request.
+func (c Context) Hostname() string {
+	ip := c.IP()
+
+	address, err := net.LookupAddr(ip)
+	if err != nil {
+		return c.Req.RemoteAddr
+	}
+
+	return address[0]
```

Comment:
Is there a chance that this ever returns an empty array? Maybe check `err != nil && len(address) > 0`?